docs/docs/when-to-use-use.md

```diff
@@ -0,0 +1,232 @@
+You could think of what the decorator does in this way:
+
+```js
+class Demo extends Component {
+  @use data = Clock;
+}
+```
+is (at runtime) transformed to:
+```js
+class Demo extends Component {
+  #data1 = invokeHelper(this, Clock);
+  
+  get data() {
+    return getValue(this.#data1);
+  }
+}
+```
+
+Note that the _actual_ return value of `resource` is an internal and private object, and interaction with that object is not supported in user-space.
+
+</details>
+
+In this example, the resource is the value of `data`:
+
+```js
+import Component from '@glimmer/component';
+import { resource, use } from 'ember-resources';
+
+const Clock = resource(() => {
+    /* ... */
+    return 2;
+});
+
+class Demo extends Component {
+    @use data = Clock;
+
+    <template>
+        {{this.time}} prints 2
+    </template>
+}
+```
+
+In this example, the `resource` is still the value returned to `data`:
+
+```js
+import Component from '@glimmer/component';
+import { resource, use } from 'ember-resources';
+
+function LocalizedClock(locale) {
+    return resource(() => {
+        /* ... */
+        return 2;
+    });    
+}
+
+class Demo extends Component {
+    @use data = LocalizedClock('en-US');
+
+    <template>
+        {{this.data}} prints 2
+    </template>
+}
+```
+
+#### "@use(ThisResource) namedThis"
+
+This variant of `@use` _only_ works if the Resource does not take arguments.
+It is the most constrained / least-flexible way to use `@use`.
+
+<details><summary>but why?</summary>
+
+This is because the left-hand side of decorators exists outside of an instance, so no instance properties may be accessed, because there is no instance.
+
+One way to think of how decorators work right now, is to "in your head", transform the code from:
+
+```js
+import Component from '@glimmer/component';
+import { resource, use } from 'ember-resources';
+
+const Clock = resource( ... );
+
+class Demo extends Component {
+    @use(Clock) time;
+}
+```
+
+To
+
+```js
+import Component from '@glimmer/component';
+import { resource, use } from 'ember-resources';
+
+const Clock = resource( ... );
+
+class Demo extends Component {}
+
+let decorator = use(Clock);
+Object.defineProperty(Demo.prototype, {
+    get() {
+        return decorator(this, 'data', /* ... */);
+    }
+}
+```
+
+</details>
+
+```js
+import Component from '@glimmer/component';
+import { resource, use } from 'ember-resources';
+
+const Clock = resource(() => 2);
+
+class Demo extends Component {
+    @use(Clock) time;
+
+    <template>
+        {{this.time}} prints 2
+    </template>
+}
+```
+
+### `use(...)`
+
+This version of `use` does not require decorators, which may be useful if you're in an environment where either you don't have the ability to use decorators, or you are composing contexts, or you want to pass a lazily evaluated pre-configured resource around your application.
+
+The returned value here is a `ReadonlyCell`, meaning you get the value of the resource via a `.current` property.
```

Comment:
I know you're looking for Docs feedback here....but I'd just thought I'd comment on the confusing DX aspect of this. It's really confusing/hard to keep track of the various things resources return without a lot of trail and error (IMHO). Is there any way to unify what's returned so that there is some consistency?

Reply:
There are only two types of return values:
- a value
- a function (aka getter)

The value -- we'll call this _Form 1_
```js
const Foo = resource(() => {
  return 'theValue';
});

<template>
  {{Foo}} prints 'theValue';
</template>
```

a function, which is a separate auto-tracking context, like a getter. This is a shorthand for the example after this one for a single value -- we'll call this _Form 2_
```js
const Foo = resource(() => {
  return () => {
    return 'theValue';
  }
});

<template>
  {{Foo}} prints 'theValue';
</template>
```

Everything else is 100% up to the user for what APIs the user wants to provide.
For example, here, we'll use an object with getters -- which are lazily evaluated. This is technically _Form 1_, but combines the concepts of _Form 1_ and _Form 2_ by building on the fundamentals of reactivity in javascript (which are presently documented nowhere :cry: I'm going to be working on this a lot very soon):

```js
const Foo = resource(() => {
  return {
    get myOwnThing() {
      return 'theValue';
    }
  }
});

<template>
  {{#let (Foo) as |theValue|}}
    {{theValue.myOwnThing}} prints 'theValue'
  {{/let}}
</template>
```

This returned value from the `resource` can be _anything_, and TypeScript correctly propagates the types, and Glint should be helping folks out when typing this -- if not, we may need to fix Glint. 

Using this example tho, we have _no need_ for _Form 2_, because getters are already lazily evaluated upon access.

We could still define it though, but behavior would be the exact same.
```js
const Foo = resource(() => {
  return () => ({
    get myOwnThing() {
      return 'theValue';
    }
  });
});

<template>
  {{#let (Foo) as |theValue|}}
    {{theValue.myOwnThing}} prints 'theValue'
  {{/let}}
</template>
```

Reply:
maybe all of this should go in another `.md` titled "what to return from your resource" or something like that

Reply:
I'm going to reply to [my other comment](https://github.com/NullVoxPopuli/ember-resources/pull/1035/files#r1384013750) with one, because they overlap. 

I'm trying to sum it up, and what seems odd to me is that how you interact with a resource is so contextually dependent.
- When you use `@use` and are in a template you can reference the resource directly
- When you use `use()` or compose a decorator with `{ use }` you need to use `.current`
- When used in plain JS you access `.value` off the resource
- Combining the last 2 means you need to use `.current.value`

Keep that straight has been a challenge. I know I probably need to broken of this, but concurrency tasks always have a predictable API no matter where it was invoked made them seem so much simpler.

Reply:
> When used in plain JS you access .value off the resource

this isn't true, there is no `.value`

> but concurrency tasks always have a predictable API no matter where it was invoked made them seem so much simpler

they have a build time transform, and are always an "object with properties" -- they have a single return value, a `TaskInstance`.

We can't have that here, because you then wouldn't be able to ergonomically use single-values as a resource.

For example:
```js

const Two = resource(() => 2);

class Demo {
  @use two = Two;
  
  <template>
    {{this.two}} 
    => you'd expect this to be '2', and not have to access this.two.value.
       .value would be the *only* property, so why would we bother with that?
       when using a decorator, we don't need to access properties, because 
       decorators allow for lazy *direct* access to values.
  </template>
}
```